### PR TITLE
Update build.rs

### PR DESCRIPTION
**Pull request by LuoZijun**
fix compile error on arm platform.

### CODE REVIEW

**Review by dignifiedquire — CHANGES_REQUESTED** [review]
this breaks because it does not respect the target architecture

**Comment by LuoZijun** [comment]
> this breaks because it does not respect the target architecture

humm, change `[target.'cfg(target_arch = "x86_64")'.build-dependencies]` to `[build-dependencies]` ?

File: build.rs [hunk 1]
```diff
@@ -1,9 +1,14 @@
+#[cfg(target_arch = "x86_64")]
 fn main() {
-    let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
```
Comment (dignifiedquire) [inline build.rs:2]: You need to use this env flag, to read out the target_arch properly. This should not be deleted

**Comment by LuoZijun** [comment]
@dignifiedquire Sorry for delay.

*Cargo.toml*:
```toml
[target.'cfg(target_arch = "x86_64")'.build-dependencies]
cc = "1.0.50"
```

*build.rs*:
```rust
fn main() {
    let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
    if target_arch == "x86_64" {
        // NOTE: crate `cc` should usable on any arch.
        //       so replace `[target.'cfg(target_arch = "x86_64")'.build-dependencies]`
        //       to `[build-dependencies]` can fix build-dependenchy on some non-x86 arch.
        cc::Build::new()
            .flag("-c")
            .file("./asm/mul_4.S")
            .compile("libff-derive-crypto.a");
    }
}
```

**Pull request closed without merging**